src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1489,10 +1488,11 @@ private void Login(ServerInfo server, TimeoutTimer timeout, string newPassword,
 
             // If the workflow being used is Active Directory Password or Active Directory Integrated and server's prelogin response
             // for FEDAUTHREQUIRED option indicates Federated Authentication is required, we have to insert FedAuth Feature Extension
-            // in Login7, indicating the intent to use Active Directory Authentication Library for SQL Server.
-            if (ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryPassword
+            // in Login7, indicating the intent to use Active Directory Authentication for SQL Server.
+            if ((ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryPassword
                 || ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryInteractive
-                || (ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryIntegrated && _fedAuthRequired))
```

Comment:
The condition seems to change a bit.

Reply:
I thought it was a bug. But now I see what it was for. Adjusted appropriately.